src/conv-core/conv-rdmadevice.h

```diff
@@ -20,24 +21,38 @@ class CmiDeviceBuffer {
   size_t cnt;
 
   // Source and destination PEs
+  /*
   int src_pe;
   int dest_pe;
+  */
 
   // Used for CUDA IPC
+  /*
   int device_idx;
   size_t comm_offset;
   int event_idx;
   cudaStream_t cuda_stream;
+  */
+
+  // Used for UCX
+  uint64_t tag;
 
   // Store the actual data for host-staged inter-node messaging (no GPUDirect RDMA)
+  /*
   bool data_stored;
   void* data;
+  */
 
-  CmiDeviceBuffer() : ptr(NULL), cnt(0), src_pe(-1), dest_pe(-1) { init(); }
+  //CmiDeviceBuffer() : ptr(NULL), cnt(0), src_pe(-1), dest_pe(-1) { init(); }
+  CmiDeviceBuffer() : ptr(NULL), cnt(0) {}
 
+  /*
   explicit CmiDeviceBuffer(const void* ptr_, size_t cnt_) : ptr(ptr_), cnt(cnt_),
     src_pe(CmiMyPe()), dest_pe(-1) { init(); }
+    */
+  explicit CmiDeviceBuffer(const void* ptr_, size_t cnt_) : ptr(ptr_), cnt(cnt_) {}
 
+  /*
```

Comment:
There are a few places in this PR, including here, where existing code is commented out. I have two suggestions for this:

1. If the content of a comment is entirely code that would otherwise compile correctly, I prefer to surround the disabled section with `#if 0` and `#endif` because this reserves the semantic of `/* block comments */` for actual comments, and also because instances of `#if 0` can be nested while block comments cannot (the closing `*/` of the inner comment will be interpreted as closing the outer one).

2. If different sections being commented out are all related by some common meaning, such as how this file has different uses of `src_pe` and `dest_pe`, instead of using a plain `#if 0` I prefer to mark them as conditional on a preprocessor definition that isn't actually defined anywhere. I'm not sure what exactly would be accurate for this case, but it could be something like `#ifdef CMIDEVICEBUFFER_OLDMETHOD` or `#ifdef CMIDEVICEBUFFER_USEIPC`.

Reply:
If the commenting out is temporary since this PR is a draft, no need to go through the trouble of the second suggestion unless you want to. I would still recommend the first.